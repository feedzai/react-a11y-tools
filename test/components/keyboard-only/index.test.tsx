/**
 * This file is open-source. This means that it can be reproduced in whole
 * or in part, stored in a retrieval system transmitted in any form, or by
 * any means electronic with my prior permission as an author and owner
 * Please refer to the terms of the license agreement in the root of the project
 *
 * (c) 2021 joaodias.me
 */
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";
import AuditInPage from "./mocks/AuditInPage";

describe("<Audit>", () => {
	it("should render the KeyboardOnly component", () => {
		const component = render(<AuditInPage />);
		expect(component).toMatchSnapshot();
	});
});
