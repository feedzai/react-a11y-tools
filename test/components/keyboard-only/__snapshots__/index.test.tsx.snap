// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Audit> should render the KeyboardOnly component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout"
      >
        <div
          data-testid="audit"
        />
        <h2>
          Sample page
        </h2>
        <a>
          Link without href
        </a>
        <button
          tabindex="1"
          type="button"
        >
          Button with higher tabindex
        </button>
        <section
          class="checkboxes-section"
        >
          <fieldset>
            <div
              class="checkbox column"
            >
              <input
                id="bell"
                name="scientist"
                type="checkbox"
                value="bell"
              />
              <label
                for="bell"
              >
                Alexander Graham Bell
              </label>
              <input
                id="curry"
                name="scientist"
                type="checkbox"
                value="curry"
              />
              <label
                for="curry"
              >
                Marie Curie
              </label>
              <input
                id="lovelace"
                name="scientist"
                type="checkbox"
                value="lovelace"
              />
              <label
                for="lovelace"
              >
                Ada Lovelace
              </label>
              <input
                id="nobel"
                name="scientist"
                type="checkbox"
                value="nobel"
              />
              <label
                for="nobel"
              >
                Alfred Nobel
              </label>
              <input
                id="pasteur"
                name="scientist"
                type="checkbox"
                value="pasteur"
              />
              <label
                for="pasteur"
              >
                Louis Pasteur
              </label>
              <input
                id="tesla"
                name="scientist"
                type="checkbox"
                value="tesla"
              />
              <label
                for="tesla"
              >
                Nikola Tesla
              </label>
            </div>
          </fieldset>
        </section>
        <section
          class="checkboxes-section"
        >
          <fieldset>
            <div
              class="checkbox column"
            >
              <input
                id="bell"
                name="scientist"
                type="checkbox"
                value="bell"
              />
              <label
                for="bell"
              >
                Alexander Graham Bell
              </label>
              <input
                id="curry"
                name="scientist"
                type="checkbox"
                value="curry"
              />
              <label
                for="curry"
              >
                Marie Curie
              </label>
              <input
                id="lovelace"
                name="scientist"
                type="checkbox"
                value="lovelace"
              />
              <label
                for="lovelace"
              >
                Ada Lovelace
              </label>
              <input
                id="nobel"
                name="scientist"
                type="checkbox"
                value="nobel"
              />
              <label
                for="nobel"
              >
                Alfred Nobel
              </label>
              <input
                id="pasteur"
                name="scientist"
                type="checkbox"
                value="pasteur"
              />
              <label
                for="pasteur"
              >
                Louis Pasteur
              </label>
              <input
                id="tesla"
                name="scientist"
                type="checkbox"
                value="tesla"
              />
              <label
                for="tesla"
              >
                Nikola Tesla
              </label>
            </div>
            <legend>
              Invalid fieldset
            </legend>
          </fieldset>
        </section>
        <section
          class="images"
        >
          <figure
            class="images__wrapper"
            id="0"
          >
            <img
              alt="Photo by Alejandro Escamilla"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/0/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1"
          >
            <img
              alt="Photo by Alejandro Escamilla"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="10"
          >
            <img
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/10/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="100"
          >
            <img
              alt="Photo by Tina Rataj"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/100/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1000"
          >
            <img
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1000/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1001"
          >
            <img
              alt="Photo by Danielle MacInnes"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1001/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1002"
          >
            <img
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1002/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1003"
          >
            <img
              alt="Photo by E+N Photographies"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1003/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1004"
          >
            <img
              alt="Photo by Greg Rakozy"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1004/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1005"
          >
            <img
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1005/640/480"
              width="640"
            />
          </figure>
          <figure
            class="images__wrapper"
            id="1006"
          >
            <img
              alt="Photo by Vladimir Kudinov"
              height="480"
              loading="lazy"
              src="https://picsum.photos/id/1006/640/480"
              width="640"
            />
          </figure>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout"
    >
      <div
        data-testid="audit"
      />
      <h2>
        Sample page
      </h2>
      <a>
        Link without href
      </a>
      <button
        tabindex="1"
        type="button"
      >
        Button with higher tabindex
      </button>
      <section
        class="checkboxes-section"
      >
        <fieldset>
          <div
            class="checkbox column"
          >
            <input
              id="bell"
              name="scientist"
              type="checkbox"
              value="bell"
            />
            <label
              for="bell"
            >
              Alexander Graham Bell
            </label>
            <input
              id="curry"
              name="scientist"
              type="checkbox"
              value="curry"
            />
            <label
              for="curry"
            >
              Marie Curie
            </label>
            <input
              id="lovelace"
              name="scientist"
              type="checkbox"
              value="lovelace"
            />
            <label
              for="lovelace"
            >
              Ada Lovelace
            </label>
            <input
              id="nobel"
              name="scientist"
              type="checkbox"
              value="nobel"
            />
            <label
              for="nobel"
            >
              Alfred Nobel
            </label>
            <input
              id="pasteur"
              name="scientist"
              type="checkbox"
              value="pasteur"
            />
            <label
              for="pasteur"
            >
              Louis Pasteur
            </label>
            <input
              id="tesla"
              name="scientist"
              type="checkbox"
              value="tesla"
            />
            <label
              for="tesla"
            >
              Nikola Tesla
            </label>
          </div>
        </fieldset>
      </section>
      <section
        class="checkboxes-section"
      >
        <fieldset>
          <div
            class="checkbox column"
          >
            <input
              id="bell"
              name="scientist"
              type="checkbox"
              value="bell"
            />
            <label
              for="bell"
            >
              Alexander Graham Bell
            </label>
            <input
              id="curry"
              name="scientist"
              type="checkbox"
              value="curry"
            />
            <label
              for="curry"
            >
              Marie Curie
            </label>
            <input
              id="lovelace"
              name="scientist"
              type="checkbox"
              value="lovelace"
            />
            <label
              for="lovelace"
            >
              Ada Lovelace
            </label>
            <input
              id="nobel"
              name="scientist"
              type="checkbox"
              value="nobel"
            />
            <label
              for="nobel"
            >
              Alfred Nobel
            </label>
            <input
              id="pasteur"
              name="scientist"
              type="checkbox"
              value="pasteur"
            />
            <label
              for="pasteur"
            >
              Louis Pasteur
            </label>
            <input
              id="tesla"
              name="scientist"
              type="checkbox"
              value="tesla"
            />
            <label
              for="tesla"
            >
              Nikola Tesla
            </label>
          </div>
          <legend>
            Invalid fieldset
          </legend>
        </fieldset>
      </section>
      <section
        class="images"
      >
        <figure
          class="images__wrapper"
          id="0"
        >
          <img
            alt="Photo by Alejandro Escamilla"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/0/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1"
        >
          <img
            alt="Photo by Alejandro Escamilla"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="10"
        >
          <img
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/10/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="100"
        >
          <img
            alt="Photo by Tina Rataj"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/100/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1000"
        >
          <img
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1000/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1001"
        >
          <img
            alt="Photo by Danielle MacInnes"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1001/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1002"
        >
          <img
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1002/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1003"
        >
          <img
            alt="Photo by E+N Photographies"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1003/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1004"
        >
          <img
            alt="Photo by Greg Rakozy"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1004/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1005"
        >
          <img
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1005/640/480"
            width="640"
          />
        </figure>
        <figure
          class="images__wrapper"
          id="1006"
        >
          <img
            alt="Photo by Vladimir Kudinov"
            height="480"
            loading="lazy"
            src="https://picsum.photos/id/1006/640/480"
            width="640"
          />
        </figure>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
